### PR TITLE
Check file conflicts in x86_64 build job

### PR DESCRIPTION
**Pull request by Chocimier**
Report packages installing same file and not marked with conflicts or replaces.

### CODE REVIEW

**Review by classabbyamp — COMMENTED** [review]
seems pretty good to me. feel free to ignore the nitpicks if you wish.

Do you have an example of the output somewhere?

File: common/travis/fetch-xtools.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 TAR=tar
 command -v bsdtar >/dev/null && TAR=bsdtar
-URL="https://github.com/leahneukirchen/xtools/archive/master.tar.gz"
+URL="https://github.com/Chocimier/xtools/archive/master.tar.gz"
```
[the remark this answers is not in the document]
Comment (Chocimier) [reply common/travis/fetch-xtools.sh]: Right, done.

File: common/scripts/lint-conflicts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+# in xlocate.
+
+export XBPS_TARGET_ARCH="$2" XBPS_DISTDIR=/hostrepo
+
+declare -A newly_built conflicts_cache providers_cache pairs owners
+repositories="--repository=$HOME/hostdir/binpkgs --repository=$HOME/hostdir/binpkgs/nonfree"
+rv=0
+
+template_exists() {
+	[ -e "srcpkgs/$1/template" ]
```
[the remark this answers is not in the document]
Comment (Chocimier) [reply common/scripts/lint-conflicts]: changed

**Comment by Chocimier** [comment]
Changed usage so it takes list of packages from local repository index and not global file. Can be easily used locally without arguments.

Test runs at <https://github.com/Chocimier/void-packages-org/actions/runs/2272244355>, <https://github.com/Chocimier/void-packages-org/actions/runs/2272208791>, <https://github.com/Chocimier/void-packages-org/actions/runs/2272194034>

File: common/scripts/lint-conflicts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+			fi
+		done < <(xbps-query $repositories -p provides -R -s "$pkgname")
+		# leading space ensures ${[]} != ''
+		providers_cache[$pkgname]=" ${providers[*]}"
+	fi
+	echo ${providers_cache[$pkgname]}
+}
+
+conflicts_of() {
+	# print list of packages that are _marked_ as confliting with given one
```
[the remark this answers is not in the document]
Comment (Chocimier) [reply common/scripts/lint-conflicts]: fixed typos

**Comment by classabbyamp** [comment]
those test runs look good to me, maybe annotations could be added à la #36950 (if they make sense to)?

**Comment by 0x5c** [comment]
> those test runs look good to me, maybe annotations could be added à la #36950 (if they make sense to)?

As an annotation without the `file` and `line` parameters yeah

**Comment by 0x5c** [comment]
> Without `file` message is not turned into comment elsewhere than logs, I do not see advantage then.

It gets a special mention in the action run summary page, and arguably the special highlighting in the logs is worth it. The message in the logs could otherwise be printed in red with ANSI sequences since GH actions render those

**Pull request merged**